File: Algorithms/plusOne.java
public class Solution {
    public int[] plusOne(int[] digits) {
        int p = 1;
        for (int i=digits.length-1; i >= 0; i--) {
            digits[i] += p;
            p = digits[i] / 10;
            digits[i] %= 10;
        }
        if (p != 0) {
            int[] new_digits = new int[digits.length+1];
            new_digits[0] = 1;
            for (int i=1; i < new_digits.length; i++){
                new_digits[i] = digits[i-1];
            }
            return new_digits;
        }
        else {
            return digits;
        }
    }
}

File: Algorithms/plusOne.py
class Solution:
    # @param digits, a list of integer digits
    # @return a list of integer digits
    def plusOne(self, digits):
        p = 1
        for i in range(len(digits)-1, -1, -1):
            digits[i] += p
            p = digits[i] / 10
            digits[i] %= 10
            if p == 0:
                return digits
        if p != 0:
            digits = [1] + digits
        return digits
